server/fleet/users.go

```diff
@@ -297,6 +297,10 @@ func (u *User) ValidatePassword(password string) error {
 }
 
 func (u *User) SetPassword(plaintext string, keySize, cost int) error {
+	if err := ValidatePasswordRequirements(plaintext); err != nil {
+		return err
+	}
```

Comment:
Ah this was a bug? (We weren't validating the password requirements on `/api/_version_/fleet/perform_required_password_reset`, we were only checking it on the UI?)

Reply:
@lucasmrod yes the validating password requirements on the backend was missing in a few other places such as cases where an admin is allowed to set/change their own (for example during the initial setup) or another user's password (such as in the create user without invite endpoint and fleetctl apply actions). 

It seemed that checks were being inconsistently implemented higher up the chain so I believe moving the check here to the SetPassword method will help to avoid this issue in the future. The one wrinkle that I found with this approach is the way we are handling the SSO users, which I see you commented on below :)